runtime/gc_vlhgc/InterRegionRememberedSet.hpp

```diff
@@ -379,12 +400,10 @@ class MM_InterRegionRememberedSet : public MM_BaseVirtual
 	MMINLINE MM_RememberedSetCard
 	convertRememberedSetCardFromHeapAddress(void* address)
 	{
-		MM_RememberedSetCard card = 0;
-#if defined(OMR_GC_COMPRESSED_POINTERS)
-		card = (MM_RememberedSetCard)((UDATA)address >> CARD_SIZE_SHIFT);
-#else
-		card = (MM_RememberedSetCard) address;	
-#endif
+		MM_RememberedSetCard card = (MM_RememberedSetCard)(UDATA)address;
+		if (compressObjectReferences()) {
+			card = (MM_RememberedSetCard)((UDATA)address >> CARD_SIZE_SHIFT);
+		}
```

Comment:
Nothing is wrong about this code however:
This optimization for Compressed Refs is based on definition:
```
#if defined (OMR_GC_COMPRESSED_POINTERS)
/* Compresses a heap address by shifting right by CARD_SIZE_SHIFT */
typedef U_32 MM_RememberedSetCard; 
#else
/* Just a heap address */
typedef UDATA MM_RememberedSetCard;
#endif /* OMR_GC_COMPRESSED_POINTERS */
```
So essentially we say Compressed Refs Shift can not be larger then CARD_SIZE_SHIFT 9 (1 to 512) and use `U_32`.
In case if non-CR an address can be any so `UDATA` must be reserved, so do not bother even to shift it.
In `OMR_GC_FULL_POINTERS` case if we set `MM_RememberedSetCard` to `UDATA` (we can do it for debugging temporary) this code is not necessary. However  the size of InterRegionRememberedSet for CR would be significantly larger. We might need to address it later by using two definitions like `MM_RememberedSetCard` and `MM_RememberedSetCardCompressed` and switch between them where necessary.

Reply:
I have not dealt with changing any of the sized types yet - I know there's still a lot of work to be done here. If it offends you, I can defer this class until the full work can be done.

Reply:
As I said nothing wrong with this code. I just trying to add note to GC reviewers to describe related (but different) problem.

Reply:
OK, let's leave this conversation open, then.

Reply:
Discussed with Dmitri, and this is likely solveable in the same way as the rest of the code - turn the constants into getters, and define them as constants for the single-reference-size builds, which should be equivalent.